code/game/objects/items/robot/robot_items.dm

```diff
@@ -545,29 +546,33 @@
 	return ..()
 
 /obj/item/borg/projectile_dampen/attack_self(mob/user)
-	if(!istype(dampening_field))
-		activate_field()
-		active = TRUE
-	else
-		deactivate_field()
-		active = FALSE
+	if(cycle_delay < world.time)
+		to_chat(user, "<span class='boldwarning'>\the [src] is still recycling its projectors!</span>")
+		return
+	cycle_delay = world.time + 20
+	active = !active
+	switch(active)
+		if(TRUE)
+			activate_field(user)
+		if(FALSE)
+			deactivate_field()
+	update_icon()
 	to_chat(user, "<span class='boldnotice'>You [active? "activate":"deactivate"] the [src].</span>")
 
 /obj/item/borg/projectile_dampen/update_icon()
 	. = ..()

```

Comment:
Update icon is one of the few places calling the base is actually a hindrance, is this necessary?